netdata-installer.sh

```diff
@@ -394,6 +388,57 @@ fi
 # replace multiple spaces with a single space
 NETDATA_CONFIGURE_OPTIONS="${NETDATA_CONFIGURE_OPTIONS//  / }"
 
+NEED_PROTOBUF=true
+
+if [ -n "${USE_SYSTEM_PROTOBUF}" ]; then
+  NEED_PROTOBUF=false
+fi
+
+if [ -n "${NETDATA_DISABLE_CLOUD}" ] && [ -n "${NETDATA_DISABLE_PROMETHEUS}" ]; then
+  NEED_PROTOBUF=false
+fi
+
+if ${NEED_PROTOBUF} && [ "$(uname -s)" = "Linux" ] && [ -f /proc/meminfo ]; then
+  mega="$((1000 * 1000))"
+
+  if [ -n "${MAKEOPTS}" ]; then
+    proc_count="$(echo ${MAKEOPTS} | grep -oE '-j *[[:digit:]]+' | tr -d '-j ')"
+  else
+    proc_count="$(find_processors)"
+  fi
+
+  target_ram="$((1500 * mega + (250 * mega * proc_count)))"
+  total_ram="$(grep MemTotal /proc/meminfo | cut -d ':' -f 2 | tr -d ' kB')"
+  total_ram="$((total_ram * 1024))"
+
+  if [ -z "${MAKEOPTS}" ]; then
+    while [ "${target_ram}" -gt "${total_ram}" ] && [ "${proc_count}" -gt 0 ]; do
+      proc_count="$((proc_count - 1))"
+      target_ram="$((1500 * mega + (250 * mega * proc_count)))"
+      MAKEOPTS="-j${proc_count}"
+    done
+
+    if [ "${proc_count}" -lt 1 ]; then
+      run_failed "Netdata needs at least ${target_ram} bytes of RAM to safely install, but this system only has ${total_ram} bytes."
+      run_failed "Insufficient RAM available for an install. Try building with the '--use-system-protobuf' flag."
+      exit 2
+    fi
+  else
+    if [ "${target_ram}" -gt "${total_ram}" ]; then
+      run_failed "Netdata needs ${target_ram} bytes of RAM to safely install, but this system only has ${total_ram} bytes."
+      run_failed "Insufficient RAM available for an install. Try reducing the number of processes used for the install using the \$MAKEOPTS variable."
+      exit 2
```

Comment:
We can get here with "-j1", the advice doesn't really work

```cmd
[pc nd_Ferroin]# MAKEOPTS="-j 1" ./netdata-installer.sh --install /opt
 FAILED  Netdata needs 1835008000 bytes of RAM to safely install, but this system only has 1073741824 bytes.

 FAILED  Insufficient RAM available for an install. Try reducing the number of processes used for the install using the $MAKEOPTS variable.
```

Perhaps 

> ... sing the \$MAKEOPTS variable or building with the '--use-system-protobuf' flag.

?

Reply:
That is an issue, but a minor one. Consider it as not worth fixing now.